pennylane/transforms/transpile.py

```diff
@@ -115,7 +115,7 @@ def stop_at(obj):
             op = list_op_copy[0]
 
             # gates which act only on one wire
-            if op.num_wires != 2:
+            if op.num_wires == 1:
```

Comment:
I think we also want to do this for gates with `num_wires > 2`, otherwise the following code crashes:
```python
def nwise(lst, k=2):
    # Function to obtain [(0,1), (1,2), (2,3), ... ], i.e. linear coupling map
    return list(zip(*[lst[i:] for i in range(k)]))

dev = qml.device("default.qubit", wires=4)

@qml.qnode(dev)
@qml.transforms.transpile(coupling_map=nwise(list(range(4))))
def circuit(param):
    qml.MultiRZ(param, wires=[0, 1, 2])
    qml.MultiRZ(param, wires=[0, 2, 3])
    qml.MultiRZ(param, wires=[0, 3, 1])
    qml.PhaseShift(param, wires=0)
    return qml.probs(wires=[0, 1, 2, 3])

print(qml.draw(circuit)(0.6))
```
Raises
```
Traceback (most recent call last):
  File "/Users/albertmitjans/Developer/pennylane/dummy/dummy.py", line 25, in <module>
    print(qml.draw(circuit)(0.6))
  File "/Users/albertmitjans/Developer/pennylane/pennylane/drawer/draw.py", line 190, in wrapper
    tapes = qnode.construct(args, kwargs)
  File "/Users/albertmitjans/Developer/pennylane/pennylane/qnode.py", line 526, in construct
    self._qfunc_output = self.func(*args, **kwargs)
  File "/Users/albertmitjans/Developer/pennylane/pennylane/tape/tape.py", line 396, in __exit__
    super().__exit__(exception_type, exception_value, traceback)
  File "/Users/albertmitjans/Developer/pennylane/pennylane/queuing.py", line 227, in __exit__
    QueuingManager.remove_active_queue()
  File "/Users/albertmitjans/Developer/pennylane/pennylane/queuing.py", line 77, in remove_active_queue
    return cls._active_contexts.pop()
IndexError: pop from empty list
```

Reply:
Could we do instead:
```suggestion
            if len(op.wires) != 2:
```

Reply:
No, because the logic there is specific to the one-qubit case.

The linked example there corresponds with the new issue that is mentioned in the PR description - supporting gates that act on more than two qubits.

Reply:
If I understood correctly, this if statement skips all gates with num_wires = 1. My question is, why not skip all gates with num_wires != 2 (or len(op.wires) != 2 to take into account gates with AnyWires)? This way the code mentioned returns the same circuit without crashing. What we could also do is raise a warning when len(op.wires) > 2 saying that qml.transpile only works for 2-qubit gates.

Reply:
Okay now I understand. It would be nice to raise an error when `len(op.wires) > 2`. Otherwise this weird error is raised, which will confuse the user.

Reply:
That's a good idea. I've added a new commit. :+1: